internal/host/host.go

```diff
@@ -206,6 +206,47 @@ func (m *Manager) HandleInstallationFailure(ctx context.Context, h *models.Host)
 	return err
 }
 
+// populateDisksEligibility updates an inventory json string by updating the eligibility
+// struct of each disk in the inventory with service-side checks for disk eligibility, in
+// addition to agent-side checks that have already been performed. The reason that some
+// checks are performed by the agent (and not the service) is because the agent has data
+// that is not available in the service.
+func (m *Manager) populateDisksEligibility(inventoryString string) (string, error) {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(inventoryString), &inventory); err != nil {
+		return "", err
+	}
+
+	for _, disk := range inventory.Disks {
+		if !disk.InstallationEligibility.Eligible && len(disk.InstallationEligibility.NotEligibleReasons) == 0 {
+			// This is a sign that the agent is using an old version that does not fill-in the eligibility struct,
+			// for backwards compatibility, pretend that the agent has decided that this disk is eligible
+			disk.InstallationEligibility.Eligible = true
+			disk.InstallationEligibility.NotEligibleReasons = make([]string, 0)
+		} else if disk.InstallationEligibility.Eligible && len(disk.InstallationEligibility.NotEligibleReasons) != 0 {
```

Comment:
How can it happen?

Reply:
A bug-free agent can never produce this, but I thought we should clear up the reasons as a server-side validation.

Reply:
i think that it will hide bugs, lets remove this part.

Reply:
Should it just be a full-on error instead, or should I just pretend like this situation can't happen?

Reply:
remove it, then it will fail because len(reasons) != 0

Reply:
Where will that failure happen?

Reply:
You already refactored this code, those it still relevant?